### ISSUE TITLE
round-change: occasional zero-value root (in messages sent)

### ISSUE THREAD

**Issue by iurii-ssv** [ssvlabs/ssv#1815]
I've noticed sometimes in logs (both production and stage) there are messages with zero-value root:
```
{"L":"DEBUG","T":"2024-10-15T20:36:21.984Z","N":"Controller.Validator","M":"🔄 got round change","pubkey":"8c1225f5b47f7aadb4e1462ecc521f27ab17dcc14a1cf92108950fe4c2745422aaf88ef768cf152879c89e2a08e5c15e","role":"PROPOSER","duty_id":"PROPOSER-e318230-s10183379-v1610715","height":10183379,"round":6,"height":10183379,"msg_round":6,"root":"0000000000000000000000000000000000000000000000000000000000000000","round-change-signers":[629]}
```

I've seen it zero valued only in messages containing `broadcasting round change message` and `got round change`, so likely round change sender is the culprit.

On the receiving side, the message root (seems like) [gets properly validated](https://github.com/ssvlabs/ssv/blob/9b70995867a215579d92c9368ae0a2227d56c5e2/protocol/genesis/qbft/instance/round_change.go#L242-L249), yet [`uponRoundChange`](https://github.com/ssvlabs/ssv/blob/9b70995867a215579d92c9368ae0a2227d56c5e2/protocol/genesis/qbft/instance/round_change.go#L17-L25) reports as if the root is "0000000000000000000000000000000000000000000000000000000000000000". 

It doesn't seem to have obvious adverse effects - which is a bit puzzling to me, could it actually mean `SignedMessage.FullData` is empty - `validRoundChangeForData` doesn't seem to be checking for this - shall we add empty-check just in case (best place to check would probably be `SignedMessage.Validate` func because it applies to all kinds of messages I guess) ?

To sum up, the issue is twofold:
- why message is sent with zero-valued root in the first place
- why the receiver doesn't seem to care (reject it when validating message contents against this zero-valued root)

Likely applies to both: pre/post Alan fork.

TODOs:
- [x]  search logs to see if happens for other message types

**Comment by Olowojaye** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by iurii-ssv** [comment]
So I checked a bunch of related things actually, my read is:

> It doesn't seem to have obvious adverse effects - which is a bit puzzling to me, could it actually mean SignedMessage.FullData is empty - validRoundChangeForData doesn't seem to be checking for this - shall we add empty-check just in case (best place to check would probably be SignedMessage.Validate func because it applies to all kinds of messages I guess) ?

For prepare & commit messages root can't be 0 or validation of root will fail (but at the same time `SignedSSVMessage.FullData` **must** be empty because, I think, the way it works is: once we have a "proposed value" there is no need to broadcast it around - operators can just agree on it and validate all QBFT messages against locally held `Instance.State.LastPreparedValue` value just by verifying message root).

Basic first validation steps are implemented here:
- [genesis](https://github.com/ssvlabs/ssv/blob/stage/message/validation/genesis/consensus_validation.go#L334-L338)
- [v2](https://github.com/ssvlabs/ssv/blob/stage/message/validation/consensus_validation.go#L105-L122)

still it seems a bit "relaxed" compared to what it could be, from what I figured ideally we want to validate this holds true (for QBFT messages):
```
ProposalMsgType: data != nil, root != 0
PrepareMsgType: data == nil, root != 0
CommitMsgType: data == nil, root != 0
RoundChangeMsgType: data == any, root == any
```
some of these conditions ^ are enforced implicitly by QBFT itself (like the case with validating message root against `Instance.State.LastPreparedValue`), but 
- it's not 100% clear if all of these are enforced implicitly
- and even if it is the case, it's probably better to fail fast and provide easy-to-read error message by validating it in `validateConsensusMessageSemantics`

> To sum up, the issue is twofold:
> 
> why message is sent with zero-valued root in the first place
> why the receiver doesn't seem to care (reject it when validating message contents against this zero-valued root)

so, answering this question for posterity,
- it seems `RoundChangeMsgType` is unique in that respect because round change might be needed regardless of whether operator managed to get hold of quorum of prepares (aka justified prepare, stored at `Instance.State.LastPreparedValue`), so for round change message root reflects the value of justified prepare (which is sent in `SignedSSVMessage.FullData` btw) which might or might not exist
- as for receiving side, zero root just means operator (the sender of round change message) didn't have any justified prepare, hence root is 0 and `SignedSSVMessage.FullData` is empty; this is fine because the leader of next round (once it's gets quorum of round changes) will propose either his own data or take the justified data from received round change message(s) and validate it (with `valCheck`, to guarantee it's sane duty data)
